document/create.go

```diff
@@ -75,90 +75,61 @@ func NewFromStruct(s interface{}) (Document, error) {
 		return nil, errors.New("expected struct or pointer to struct")
 	}
 
-	return structDocument{ref: ref}, nil
+	return newFromStruct(ref)
 }
 
-type structDocument struct {
-	ref reflect.Value
-}
-
-var _ Document = (*structDocument)(nil)
-
-func (s structDocument) Iterate(fn func(field string, value Value) error) error {
-	l := s.ref.NumField()
-
-	tp := s.ref.Type()
+func newFromStruct(ref reflect.Value) (Document, error) {
+	var fb FieldBuffer
+	l := ref.NumField()
+	tp := ref.Type()
 
 	for i := 0; i < l; i++ {
-		sf := tp.Field(i)
-		if sf.PkgPath != "" {
+		f := reflect.Indirect(ref.Field(i))
+		if !f.IsValid() {
 			continue
 		}
 
-		var name string
-		if gtag, ok := sf.Tag.Lookup("genji"); ok {
-			if gtag == "-" {
-				continue
-			}
-
-			name = gtag
-		} else {
-			name = strings.ToLower(sf.Name)
-		}
-
-		f := s.ref.Field(i)
+		sf := tp.Field(i)
 
-		v, err := NewValue(f.Interface())
-		if err != nil {
-			if err.(*ErrUnsupportedType) != nil {
-				continue
+		switch f.Kind() {
```

Comment:
A switch with two cases is uncommon, how about having just an `if` with a `return` in it?

Reply:
There is no return inside the switch we are in a `for` loop. We can still do an `if-else` statement.

Reply:
You can use `continue` instead or `return` in that case